openai/tests/test_client.py

```diff
@@ -1,7 +1,52 @@
+import os
 import pytest
-
+import functools
 import openai.client
 
+# Live tests
+API_TYPE = ["azure", "openai", "azuredefault"]
+API_BASE = os.environ["AZURE_API_BASE"]
+AZURE_API_KEY = os.environ["AZURE_KEY"]
+OPENAI_API_KEY = os.environ["OPENAI_KEY"]
+API_VERSION  = "2023-06-01-preview"
+COMPLETION_MODEL = "text-davinci-003"
+CHAT_COMPLETION_MODEL = "gpt-35-turbo"
+CHAT_COMPLETION_MODEL_OPENAI = "gpt-3.5-turbo"
+EMBEDDINGS_MODEL = "text-embedding-ada-002"
+IMAGE_PATH = ""
+MASK_IMAGE_PATH = ""
+
+
+def configure_client(f):
+    @functools.wraps(f)
+    def wrapper(*args, **kwargs):
+        api_type = kwargs.pop("api_type")
+        if api_type == "azure":
+            client = openai.client.OpenAIClient(
+                api_base=API_BASE,
+                auth=AZURE_API_KEY,
+                api_version=API_VERSION,
+                backend="azure"
+            )
+        elif api_type == "azuredefault":
+            api_type = "azure"
+            client = openai.client.OpenAIClient(
+                api_base=API_BASE,
+                auth="azuredefault",
+                api_version=API_VERSION,
+                backend="azure"
+            )
+        elif api_type == "openai":
+            client = openai.client.OpenAIClient(
+                auth=OPENAI_API_KEY,
+                backend="openai"
+            )
+        kwargs = {"client": client}
```

Comment:
We are silently dropping any `kwargs` here. Is that intentional? If so, I'd explicitly pass in `client` to the wrapped `f` on the next line and not do a assign-new-value-to-`kwargs`-and-spread-that-in.

Reply:
Simplified it. Didn't realize I could pass the values of parametrize to a fixture, but pytest lets you do it cleanly.